### PR TITLE
Delaware Additional Personal Credits

### LINKED ISSUE #2456: Delaware Additional Personal Credits
Delaware Additional Personal Credits
Reference: 
2022 - https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
2021 - https://revenuefiles.delaware.gov/2021/PIT-RES_TY21_2021-01_Instructions.pdf

### PR DESCRIPTION
**Pull request by Kiki-67**
Fixes #2456

### Summary
🆕📝🧮

This pull request adds a new feature to the policy engine: the Delaware additional personal credits for aged filers. It introduces new parameters, variables, and tests for the deduction formula, and updates the changelog entry accordingly. The files affected are `changelog_entry.yaml`, `age_eligibility.yaml`, `aged.yaml`, `de_aged_additional_personal_credits.yaml`, and `de_aged_additional_personal_credits.py`.

> _`deduction` added_
> _Delaware credits for aged_
> _autumn of their lives_

### Walkthrough
*  Add a new feature to the policy engine: the Delaware additional personal credits for aged filers ([link](https://github.com/PolicyEngine/policyengine-us/pull/2458/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8L3-R3))
*  Define the age eligibility parameter for the credit, which requires filers to be at or above 65 years old ([link](https://github.com/PolicyEngine/policyengine-us/pull/2458/files?diff=unified&w=0#diff-bd7d0dade83b3fa8f531af9572e6b06b2e772a6a25d70a0afa1ab84180c37aa0R1-R13))
*  Define the aged deduction amount parameter for the credit, which allows eligible filers to deduct $110 per person from their taxable income ([link](https://github.com/PolicyEngine/policyengine-us/pull/2458/files?diff=unified&w=0#diff-e622bf22bc2f48aa1c16d87be7fe030597b5fe1c13e90e1323f51900858079b3R1-R15))
*  Implement the deduction variable for the credit, which applies the formula based on the age and eligibility parameters ([link](https://github.com/PolicyEngine/policyengine-us/pull/2458/files?diff=unified&w=0#diff-c9ecd238a6ccf4a39a8808cfdfa167ec6db6f37a53b1f93710a52b51561d8d67R1-R20))
*  Add test cases for the deduction variable, covering different scenarios of age and filing status for the tax unit ([link](https://github.com/PolicyEngine/policyengine-us/pull/2458/files?diff=unified&w=0#diff-05f1861f6cbb76f8259054bd94cb09d359254afc186336b662ea6750b8f6b0acR1-R44))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:1]: need legal code reference

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+description: Delaware provides the additional personal credits amount for aged filers.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml:1]:
```suggestion
description: Delaware provides this additional personal credits amount for aged filers.
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+values:
+  2021-01-01: 110
+  2022-01-01: 110
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml:5]:
```suggestion
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  unit: /1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml:7]:
```suggestion
  unit: currency-USD
```

File: policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credits.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class de_aged_personal_credits_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Delaware aged additional personal credits deduction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credits.py:7]:
```suggestion
    label = "Delaware aged additional personal credits"
```

File: policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credits.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class de_aged_personal_credits_deduction(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credits.py:4]:
```suggestion
class de_aged_personal_credits(Variable):
```

File: policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credits.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credits.py:1]: rename file to de_aged_additional_personal_credits.py

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+description: Delaware provides the aged additional personal credits amount to filers at or above this age.
+values:
+  2021-01-01: 65
+  2022-01-01: 65
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:4]:
```suggestion
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+description: Delaware provides this additional personal credits amount for aged filers.
+
+values:
+  2021-01-01: 110
+metadata:
+  unit: currency-USD
+  period: year
+  name: Delaware_additional_personal_credits_deduction_amount_aged
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml:8]:
```suggestion
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  label: Delaware aged additional personal credits deduction amount
+  reference:
+    - title: PIT-RES_TY22_2022-02_Instruction Page 8 Line 26b.
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
+    - title: PIT-RES_TY22_2021-02_Instruction Page 8 Line 26b.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml:13]: add legal code

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  unit: year
+  period: year
+  label: Delaware additional aged personal credits age eligibility 
+  reference:
+    - title: PIT-RES_TY22_2022-02_Instruction Page 8 Line 26b.
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
+    - title: PIT-RES_TY22_2021-02_Instruction Page 8 Line 26b.
+      href: https://revenuefiles.delaware.gov/2021/PIT-RES_TY21_2021-01_Instructions.pdf
+    - title: Delaware Code Online Chapter 11 § 1110.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:13]:
```suggestion
    - title: Delaware Code Online Chapter 11 § 1110(b)(2)
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+      href: https://delcode.delaware.gov/title30/c011/sc02/index.html
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:14]:
```suggestion
      href: https://delcode.delaware.gov/title30/c011/sc02/index.html#1110
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
+description: Delaware provides the aged additional personal credits amount to filers at or above this age.
+values:
+  2021-01-01: 65
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:3]: Per the law and tax forms. Please be careful.
```suggestion
  2021-01-01: 60
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  unit: year
+  period: year
+  label: Delaware additional aged personal credits age eligibility 
+  reference:
+    - title: PIT-RES_TY22_2022-02_Instruction Page 8 Line 26b.
+      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:10]:
```suggestion
      href: https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf#page=8
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:1]: combine this and the value into a single scale parameter

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml [hunk 13]
```diff
@@ -0,0 +1,23 @@
+description: Delaware provides the aged additional personal credits amount to filers at or above this age.
+
+metadata:
+  type: single_amount
+  unit: year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:5]:
```suggestion
  threshold_unit: currency-USD
  amount_unit: year
  
```

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/aged.yaml:1]: delete

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:1]: Rename `deductions/age_eligibility.yaml` to `credits/personal/aged.yaml` (this is not about eligibility only)

File: policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml [hunk 14]
```diff
@@ -0,0 +1,25 @@
+description: Delaware provides the aged additional personal credits amount to filers at or above this age.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/de/tax/income/deductions/age_eligibility.yaml:1]:
```suggestion
description: Delaware provides this aged additional personal credit amount to the head and spouse of a filing unit based on their age.
```

File: policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credites.py [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+    label = "Delaware aged additional personal credits"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.DE
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.de.tax.income.deductions
+
+        age_head = tax_unit("age_head", period)
+        head_eligible = p.age_eligibility.calc(age_head)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credites.py:16]: and spouse
```suggestion
        head_amount = p.age_eligibility.calc(age_head)
```

File: policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credites.py [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Delaware aged additional personal credits"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.DE
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.de.tax.income.deductions
+
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/de/tax/income/deductions/de_aged_additional_personal_credites.py:14]:
```suggestion
        # Legal code says "An additional $110 in the case of each resident person age 60 or over."
        # Tax form limits it to heads and spouses, not elderly dependents.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**